src/unix/linux/android/audio.c

```diff
@@ -10,20 +10,22 @@
 
 #include <aaudio/AAudio.h>
 
-#define AUDIO_SAMPLE_SIZE sizeof(int16_t)
+#define AUDIO_SAMPLE_SIZE(fmt) ((fmt) == MTY_AUDIO_SAMPLE_FORMAT_FLOAT ? sizeof(float) : sizeof(int16_t))
 
 #define AUDIO_BUF_SIZE(ctx) \
-	((ctx)->sample_rate * (ctx)->channels * AUDIO_SAMPLE_SIZE)
+	((ctx)->sample_rate * (ctx)->channels * AUDIO_SAMPLE_SIZE((ctx)->sample_format))
 
 struct MTY_Audio {
 	AAudioStreamBuilder *builder;
 	AAudioStream *stream;
 
 	MTY_Mutex *mutex;
+	MTY_AudioSampleFormat sample_format;
 	uint32_t sample_rate;
```

Comment:
Add sample size here, so it just needs to be evaluated once.
```suggestion
	MTY_AudioSampleFormat sample_format;
	size_t sample_size;
	uint32_t sample_rate;
```

Where the it will be: `sample_size  = channels * AUDIO_SAMPLE_SIZE(ctx->sample_format)`

Reply:
Done, let me know if you're happy with the changes.